core/operation/src/main/java/uk/gov/gchq/gaffer/jobtracker/JobDetail.java

```diff
@@ -60,36 +66,37 @@ public JobDetail(final JobDetail oldJobDetail, final JobDetail newJobDetail) {
         }
     }
 
-    public JobDetail(final String jobId, final String userId, final OperationChain<?> opChain, final JobStatus jobStatus, final String description) {
-        this(jobId, null, userId, opChain, jobStatus, description);
+    public JobDetail(final String jobId, final User user, final OperationChain<?> opChain, final JobStatus jobStatus, final String description) {
```

Comment:
This will change the public API which would mean a breaking change. We've committed not to introduce breaking changes in version 1.x. Would it be possible to revert these and add some new constructors.
```suggestion
    @Deprecated
    public JobDetail(final String jobId, final String userId, final OperationChain<?> opChain, final JobStatus jobStatus, final String description) {
        this(jobId, null, new User(userId), opChain, jobStatus, description);
    }
    // other constructors following same pattern
    public JobDetail(final String jobId, final String parentJobId, final User user, final OperationChain<?> opChain, final JobStatus jobStatus, final String description) {   
         // do stuff
     }
```

Reply:
Re-instated deleted code and marked @Deprecated